Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -153,14 +153,22 @@ fileprivate extension TrialOrIntroPriceEligibilityChecker {
 
 }
 
-private extension TrialOrIntroPriceEligibilityChecker {
+internal extension TrialOrIntroPriceEligibilityChecker {
 
     @available(iOS 11.2, macOS 10.13.2, macCatalyst 13.0, tvOS 11.2, watchOS 6.2, *)
-    func productsWithIntroOffers(productIdentifiers: [String], completion: @escaping ReceiveIntroEligibilityBlock) {
+    func mapProductsWithIntroEligibility(productIdentifiers: [String],
+                                         completion: @escaping ReceiveIntroEligibilityBlock) {
         self.productsManager.products(withIdentifiers: Set(productIdentifiers)) { products in
             let eligibility: [(String, IntroEligibility)] = Array(products.value ?? [])
-                .filter { $0.introductoryDiscount != nil }
-                .map { ($0.productIdentifier, IntroEligibility(eligibilityStatus: .eligible)) }
+                .map({ storeProduct in
```

Comment:
`.map({` -> `map {`

Reply:
I'm mad the linter didn't catch this 🙄

Reply:
There's probably a rule for it

Reply:
This is the rule: https://realm.github.io/SwiftLint/trailing_closure.html
Unfortunately it forces using the trailing closure syntax everywhere, which we probably don't want (it's nice to be explicit with the parameter name sometimes).